src/backends/cpu.jl

```diff
@@ -242,10 +242,12 @@ Base.eltype(a::ScratchArray) = eltype(a.data)
     return ScratchArray{length(Dims)}(MArray{__size((Dims..., __groupsize(ctx.metadata)...)), T}(undef))
 end
 
-Base.@propagate_inbounds function Base.getindex(A::ScratchArray, I...)
+Base.@propagate_inbounds function Base.getindex(A::ScratchArray{N}, I...) where {N}
+    return @view A.data[ntuple(_ -> Colon(), Val(N))..., I...]
```

Comment:
I find `@view` hard to read can you use `view` directly?

Reply:
Done.